config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -1254,7 +1273,7 @@ public void setBearerTokenConverter(ServerAuthenticationConverter bearerTokenCon
 		 * Configures JWT Resource Server Support
 		 */
 		public class JwtSpec {
-			private ReactiveAuthenticationManager authenticationManager;
+			private ReactiveAuthenticationManagerResolver<ServerHttpRequest> authenticationManagerResolver;
```

Comment:
The tests for these ought to go into [OAuth2ResourceServerSpecTests](https://github.com/spring-projects/spring-security/blob/master/config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java).

Reply:
@jzheaux struggling with setting up the `configuration` for this one. I have played around with variation of the following and the results I get are 

- No qualifying bean for `ReactiveJwtDecoder` and when I declare that when stubs then it results in status 200 ok ignoring the authentication manager and resolver 😕 
- Without `ReactiveJwtDecoder` I get `500 error` 

```
@EnableWebFlux
@EnableWebFluxSecurity
static class CustomAuthenticationManagerResolverConfig {
	@Bean
	SecurityWebFilterChain springSecurity(ServerHttpSecurity http) throws Exception {
		// @formatter:off
		http
			.authorizeExchange()
				.pathMatchers("/**/message/**").hasAnyAuthority("SCOPE_message:read")
				.and()
			.oauth2ResourceServer()
				.authenticationManagerResolver(authenticationManagerResolver())
				.jwt();
		// @formatter:on
		return http.build();
	}

	@Bean
	ReactiveAuthenticationManagerResolver<ServerHttpRequest> authenticationManagerResolver() {
		return mock(ReactiveAuthenticationManagerResolver.class);
	}
	@Bean
	ReactiveAuthenticationManager authenticationManager() {
		return mock(ReactiveAuthenticationManager.class);
	}
}
```
if I chain the `authenticationManager` after `jwt()` as follow 

```
.oauth2ResourceServer()
		.authenticationManagerResolver(authenticationManagerResolver())
	        .jwt()
		.authenticationManager(authenticationManager())
``` 
then it works but, then how is that a test of `authenticationManagerResolver` then? 🤔

Reply:
So, you wouldn't supply both `jwt()` and `authenticationManagerResolver()` together. (There is a separate ticket to clean this up).

If you do:

```java
@EnableWebFlux
@EnableWebFluxSecurity
static class CustomAuthenticationManagerResolverConfig {
	@Bean
	SecurityWebFilterChain springSecurity(ServerHttpSecurity http) throws Exception {
		// @formatter:off
		http
			.authorizeExchange()
				.pathMatchers("/**/message/**").hasAnyAuthority("SCOPE_message:read")
				.and()
			.oauth2ResourceServer()
				.authenticationManagerResolver(authenticationManagerResolver());
		// @formatter:on
		return http.build();
	}

	@Bean
	ReactiveAuthenticationManagerResolver<ServerHttpRequest> authenticationManagerResolver() {
		return mock(ReactiveAuthenticationManagerResolver.class);
	}
}
```

then does it work?

Reply:
I am certain that I have given that a try and came across an issue. I have tried a few variations of the configurations and got errors messages such as 
- A `404` error
- An error complaining about `ReactiveJwtDecoder` not being there, although I feel like this is due to use of `jwt()` 
- A `500` null pointer exception in some `reactive` lamda 

I can give it another try later in the day and push my tests as well as share the error messages! :slightly_smiling_face: